docs/technical_reference/unit_models/uv_aop.rst

```diff
@@ -0,0 +1,80 @@
+
+Ports
+-----
+
+The model provides two ports (Pyomo notation in parentheses):
+
+* Inlet port (inlet)
+* Outlet port (outlet)
+
+Sets
+----
+.. csv-table::
+   :header: "Description", "Symbol", "Indices"
+
+   "Time", ":math:`t`", "[0]"
+   "Phases", ":math:`p`", "['Liq']"
+   "Components", ":math:`j`", "['H2O', 'NDMA', ...]"
+
+Degrees of Freedom and Variables
+--------------------------------
+Aside from the inlet feed state variables (i.e., temperature, pressure, component mass flowrates),
+the UV AOP model has at least an additional 5 degrees of freedom that
+the user must specify. The table below gives an outline of these.
+
+.. csv-table::
+   :header: "Description", "Symbol", "Variable Name", "Index", "Units"
+
+   "Inlet temperature", ":math:`T`", "temperature", "[t]", ":math:`\text{K}`"
+   "Inlet pressure", ":math:`P_{in}`", "pressure", "[t]", ":math:`\text{Pa}`"
+   "Outlet pressure", ":math:`P_{out}`", "pressure", "[t]", ":math:`\text{Pa}`"
+   "Mass flowrate of components", ":math:`M_j`", "flow_mass_phase_comp", "[t, 'Liq', j]", ":math:`\text{kg/s}`"
+   "Inlet volumetric flowrate", ":math:`F_{in}`", "flow_vol", "[t]", ":math:`\text{m}^3\text{/s}`"
+   "Inactivation rate coefficient", ":math:`k`", "inactivation_rate", "['Liq', j]", ":math:`\text{m}^2\text{/J}`"
+   "Pseudo-first order rate constant", ":math:`k_0`", "rate_constant", "['Liq', j]", ":math:`\text{s}^{-1}`"
+   "UV dose", ":math:`D`", "uv_dose", None, ":math:`\text{J/}\text{m}^2`"
+   "Average intensity of UV light", ":math:`I`", "uv_intensity", None, ":math:`\text{J/}\text{m}^2\text{/s}`"
+   "Exposure time of UV light", ":math:`t`", "exposure_time", None, ":math:`\text{s}`"
+   "Electricity demand of components", ":math:`E_j`", "electricity_demand_phase_comp", "[t, 'Liq', j]", ":math:`\text{W}`"
+   "Electricity efficiency per log order reduction (EE/O)", ":math:`EE/O_j`", "electrical_efficiency", "[t, 'Liq', j]", ":math:`\text{J/}\text{m}^3`"
+   "Lamp efficiency", ":math:`\eta`", "lamp_efficiency", None, None
+
+**Users must provide values for and 'fix' these variables to solve the model with DOF=0. However, users may also leave variables unfixed for optimization purposes.**
+
+**NOTE: Variables for 'temperature', 'pressure', and 'flow_mass_phase_comp' come from the associated property package as state variables and are accessed via {port_name}.{state_var_name}**
+
+Equations and Relationships
+---------------------------
+
+.. csv-table::
+   :header: "Description", "Equation"
+
+   "UV dose", ":math:`D = I \cdot t`"
+   "Pseudo-first order rate constant", ":math:`k_0 = I \cdot k`"
+   "Solvent mass balance", ":math:`M_{\text{H2O},out} = M_{\text{H2O},in}`"
+   "Solute mass balance", ":math:`M_{j,out} = M_{j,in} \cdot \exp(D \cdot k)`"
+   "Electricity demand", ":math:`E_j = EE/O_j \cdot F_{in} \cdot \log(M_{j,in} / M_{j,out}) / \eta`"
+
```

Comment:
Maybe it would be best to include the Class Documentation block of code here, find an example in the Reverse Osmosis (0D) page. As a standalone documentation, I am confused on whether this applies to both the UV and UV-AOP models in WaterTAP and this code block can insert an automated link to the model class that you are describing.

Reply:
Looks like yours is listed as `watertap.unit_models.zero_order.uv_zo` and `watertap.unit_models.zero_order.uv_aop_zo` so add whichever (or both) this documentation is referencing

Reply:
Suggesting this as the link didn't generate in the most recent commit, 2948a1b

Reply:
Yeah, I've aded the Class Documentation block as you suggested.

Reply:
Excuse me, maybe it should be `watertap.unit_models.ultraviolet`. It should be linking to the "watertap.unit_models package" page when linked properly. I believe its dot notation based on the name of the .py file in the unit_models folder.

Reply:
@hunterbarber No worry, I did add `watertap.unit_models.uv_aop`, it should be correct. The uv_zo and uv_aop_zo are previous WaterTAP3 models.

Reply:
Ah, I see the most recent PR now, sorry! Should be good after both PRs get merged.